analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -30,6 +30,8 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn
 {
     internal class RoslynSymbolicExecution
     {
+        internal const int MaxStepCount = 2000;
```

Comment:
Is now the right time to do this? Shouldn't we add this check only after we measure the performance once the engine is mature enough?

Reply:
We need to add safety limitations as soon as possible to avoid unexpected performance explosions. Once the engine is mature enough, we can measure it and adjust this to a better value.